core/src/avm2/globals/flash/utils.rs

```diff
@@ -140,6 +140,56 @@ pub fn escape_multi_byte<'gc>(
     Ok(v.into())
 }
 
+fn handle_percent<I>(it: &mut I) -> Option<u8>
+where
+    I: Iterator<Item = char>,
+{
+    let mut d = it.next()?.to_digit(16)?;
+    d = d * 16 + it.next()?.to_digit(16)?;
```

Comment:
nit: This can be cleaned up a bit (no `mut`, bitwise instead of arithmetic operations):
```rs
let high = it.next()?.to_digit(16)? as u8;
let low = it.next()?.to_digit(16)? as u8;
Some(low | (high << 4))
```

Reply:
Done.